src/main.rs

```diff
@@ -1,116 +1,129 @@
+                let bytes_written = nix::unistd::write(self.master_pty, &buffer[..bytes_read])?;
 
-                    //flush it
-                    master_file.flush()?;
-                    if bytes_read != bytes_written {
-                        bail!("partial failed read[{}]/write[{}] (masterFd)", bytes_read, bytes_written);
-                    }
+                if bytes_read != bytes_written {
+                    bail!("partial failed read[{}]/write[{}] (masterFd)", bytes_read, bytes_written);
                 }
+            }
 
-                // if the pseudo-terminal master has input available, this program reads some of that
-                // input and writes it to the terminal and file
-                if in_fds.contains(fork_result.master) {
-                    let bytes_read_result = master_file.read(&mut buffer);
-
-                    // IO error here is a happy case since the child process might have died
-                    // hide it by returning OK
-                    let bytes_read = match bytes_read_result {
-                        Ok(bytes_read) => bytes_read,
-                        Err(_) => return Ok(())
-                    };
-
-                    let bytes_written = nix::unistd::write(stdout, &buffer[..bytes_read])?;
-                    if  bytes_written != bytes_read {
-                        bail!("partial failed read[{}]/write[{}] (stdout)", bytes_read, bytes_written);
-                    }
-
-                    let bytes_written = output_file.write(&buffer[..bytes_read])?;
-                    if bytes_written != bytes_read {
-                        bail!("partial failed read[{}]/write[{}] (output file)", bytes_read, bytes_written);
-                    }
+            // if the pseudo-terminal master has input available, this program reads some of that
+            // input and writes it to the terminal and file
+            if in_fds.contains(self.master_pty) {
+                let bytes_read_result = nix::unistd::read(self.master_pty, &mut buffer);
+
+                // IO error here is a happy case since the child process might have died
+                // hide it by returning OK
+                let bytes_read = match bytes_read_result {
+                    Ok(bytes_read) => bytes_read,
+                    Err(_) => return Ok(())
+                };
+
+                let bytes_written = nix::unistd::write(self.stdout, &buffer[..bytes_read])?;
+                if  bytes_written != bytes_read {
+                    bail!("partial failed read[{}]/write[{}] (stdout)", bytes_read, bytes_written);
                 }
 
+                let bytes_written = self.typescript.write(&buffer[..bytes_read])?;
+                if bytes_written != bytes_read {
+                    bail!("partial failed read[{}]/write[{}] (output file)", bytes_read, bytes_written);
+                }
             }
         }
     }
+}
 
-    Ok(())
+#[derive(structopt::StructOpt)]
+#[structopt(name = "script")]
```

Comment:
@markrwilliams you love macros or whatever these are.
I'm surprised you haven't gone full into Java annotation programming ;)